src/main/java/com/tterrag/registrate/providers/RegistrateAdvancementProvider.java

```diff
@@ -1,86 +1,91 @@
 package com.tterrag.registrate.providers;
 
-import java.io.IOException;
-import java.nio.file.Path;
-import java.util.HashSet;
-import java.util.Objects;
-import java.util.Set;
-import java.util.function.Consumer;
-
-import javax.annotation.Nullable;
-
+import com.google.common.collect.Lists;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.tterrag.registrate.AbstractRegistrate;
-
 import lombok.extern.log4j.Log4j2;
+
 import net.minecraft.advancements.Advancement;
+import net.minecraft.core.HolderLookup;
 import net.minecraft.data.CachedOutput;
-import net.minecraft.data.DataGenerator;
 import net.minecraft.data.DataProvider;
-import net.minecraft.data.HashCache;
+import net.minecraft.data.PackOutput;
 import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraftforge.fml.LogicalSide;
 
+import javax.annotation.Nullable;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
 @Log4j2
 public class RegistrateAdvancementProvider implements RegistrateProvider, Consumer<Advancement> {
 
     private static final Gson GSON = (new GsonBuilder()).setPrettyPrinting().create();
 
     private final AbstractRegistrate<?> owner;
-    private final DataGenerator generator;
+    private final PackOutput packOutput;
+    private final CompletableFuture<HolderLookup.Provider> registries;
+    private final List<CompletableFuture<?>> advancementsToSave = Lists.newArrayList();
 
-    public RegistrateAdvancementProvider(AbstractRegistrate<?> owner, DataGenerator generatorIn) {
+    public RegistrateAdvancementProvider(AbstractRegistrate<?> owner, PackOutput packOutput, CompletableFuture<HolderLookup.Provider> registriesLookup) {
         this.owner = owner;
-        this.generator = generatorIn;
+        this.packOutput = packOutput;
+        registries = registriesLookup;

```

Comment:
These assignments (in all data provider classes) should follow the pattern above (with `this.`).